railties/lib/rails/generators/primary_file_helpers.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Rails
+  module Generators
+    module PrimaryFileHelpers # :nodoc:
+      extend ActiveSupport::Concern
+
+      included do
+        class_option :editor, type: :string, aliases: "-e", lazy_default: ENV["EDITOR"],
+                              required: false, banner: "editor",
+                              desc: "Open generated primary file in the specified editor (Default: $EDITOR)"
```

Comment:
Should this really be an `--editor` option or a `--edit` option? Because we don't allow editor customization for the `credentials:edit` or `secrets:edit` commands, there `$EDITOR` is assumed.

Reply:
I don't think we indent arguments like this. I'd just do:

```ruby
class_option :editor, type: :string, …
  required: false, …
```